### PR TITLE
Adding Advanced Threat Protection and Vulnerability Assessment cmdlets

### PR DESCRIPTION
**Pull request by yaakoviyun**
Adding Advanced Threat Protection and Vulnerability Assessment cmdlets

## Description

Using the Vulnerability Assessment for Azure SQL database offering, customers can scan their databases with a predefined set of rules (security checks) and get a full picture of their SQL vulnerabilities and have the ability to remediate each one.
More info can be found at Blog - https://docs.microsoft.com/en-us/azure/sql-database/sql-vulnerability-assessment

Design review can be found at: Azure/azure-powershell-cmdlet-review-pr#78

This PR is a continuation of : https://github.com/Azure/azure-powershell/pull/6442
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by yaakoviyun** [comment]
@MiYanni  - this is the PR with one clean check in

**Comment by yaakoviyun** [comment]
@yoavfr , @ronitr

**Comment by yaakoviyun** [comment]
@MiYanni - the builds are successful.

Can we move ahead with the PR merge?

File: src/ResourceManager/Sql/Commands.Sql/ChangeLog.md [hunk 1]
```diff
@@ -28,6 +28,12 @@
 * Clarified User-Defined Restore Points for SQLDW in New-AzureRmSqlDatabaseRestorePoint help
 * Fixed formatting of OutputType in help files
 * Updated documentation of -ComputeGeneration parameter in several cmdlets
+* Adding Server Advanced Threat Protection support with the following cmdlets:
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql/ChangeLog.md]: @yaakoviyun These changelog entries need to be moved up under the `Current Release` section.

File: src/ResourceManager/Sql/Commands.Sql/ThreatDetection/Cmdlet/SetAzureSqlServerThreatDetection.cs [hunk 2]
```diff
@@ -52,7 +52,6 @@ public class SetAzureSqlServerThreatDetection : SqlServerThreatDetectionCmdletBa
         /// Gets or sets the name of the storage account to use.
         /// </summary>
         [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the storage account")]
-        [ValidateNotNullOrEmpty]
```
Comment (MiYanni) [inline src/ResourceManager/Sql/Commands.Sql/ThreatDetection/Cmdlet/SetAzureSqlServerThreatDetection.cs]: Why was this removed? Would someone provide a storage account name (which is an optional parameter) that is null or empty?

Comment (yaakoviyun) [reply src/ResourceManager/Sql/Commands.Sql/ThreatDetection/Cmdlet/SetAzureSqlServerThreatDetection.cs]: The backend now support setting an empty storage account name, but I think I'll revert it and let the threat detection folks define it more properly

**Comment by yaakoviyun** [comment]
@MiYanni  - I've fixed all your comments.
Waiting for your approval and final merge.

@yoavfr and @ronitr for awareness

**Comment by yaakoviyun** [comment]
@ayeletshpigelman

**Comment by MiYanni** [comment]
@maddieclayton Can you merge this when it passes the build?

**Comment by maddieclayton** [comment]
@MiYanni Should I take one more look over the code? Otherwise, can you approve?
